### PR TITLE
New FOSSology UI with bootstrap 4

### PR DESCRIPTION
**Pull request by shaheemazmalmmd**
## Description

Include new bootstrap 4 UI for fossology. 

### Changes

* Change login, browse, single file, tree view, search and copyright pages. 

## How to test

* End to end testing of each and every feature.

### CODE REVIEW

**Comment by GMishx** [comment]
Some points to cover:
1. Adding a user decision or changing license text (making file in WIP mode) did not add the RED border to the decision group
image
2. Clearing 23/23 files (100%) did not progress the bar to 100% width
image
3. Bulk Modal can use more width to have the "Ignore Irrelevant files" in the same line as of the checkbox or the whole thing can move to next line.
image
4. Clicking on "Clean Text" in bulk modal closes it.
5. Clicking [Edit] on a folder
image
6. Please change the button class in Conf, Export Lists, Spasht, "Jobs > Schedule Jobs", pages
7. Links change to text color, causing confusion whenever they are inside Tabs (Please check copyright and Author list). Working everywhere else (like ECC).
image

**Comment by GMishx** [comment]
Please check following notices in Apache:
```
PHP Notice:  Undefined variable: filesCleared in /usr/local/share/fossology/www/ui/ui-clearing-view.php on line 324, referer: http://localhost/repo/?mod=license&upload=235&folder=20&item=493955&show=quick
PHP Stack trace:, referer: http://localhost/repo/?mod=license&upload=235&folder=20&item=493955&show=quick
PHP   1. {main}() /usr/local/share/fossology/www/ui/index.php:0, referer: http://localhost/repo/?mod=license&upload=235&folder=20&item=493955&show=quick
PHP   2. ClearingView->execute() /usr/local/share/fossology/www/ui/index.php:64, referer: http://localhost/repo/?mod=license&upload=235&folder=20&item=493955&show=quick
PHP   3. ClearingView->getResponse() /usr/local/share/fossology/www/ui/ui-clearing-view.php:137, referer: http://localhost/repo/?mod=license&upload=235&folder=20&item=493955&show=quick
PHP   4. ClearingView->Output() /usr/local/share/fossology/lib/php/Plugin/FO_Plugin.php:385, referer: http://localhost/repo/?mod=license&upload=235&folder=20&item=493955&show=quick
PHP Notice:  Undefined variable: filesCleared in /usr/local/share/fossology/www/ui/ui-clearing-view.php on line 324, referer: http://localhost/repo/?mod=view-license&upload=235&item=494099
PHP Stack trace:, referer: http://localhost/repo/?mod=view-license&upload=235&item=494099
PHP   1. {main}() /usr/local/share/fossology/www/ui/index.php:0, referer: http://localhost/repo/?mod=view-license&upload=235&item=494099
PHP   2. ClearingView->execute() /usr/local/share/fossology/www/ui/index.php:64, referer: http://localhost/repo/?mod=view-license&upload=235&item=494099
PHP   3. ClearingView->getResponse() /usr/local/share/fossology/www/ui/ui-clearing-view.php:137, referer: http://localhost/repo/?mod=view-license&upload=235&item=494099
PHP   4. ClearingView->Output() /usr/local/share/fossology/lib/php/Plugin/FO_Plugin.php:385, referer: http://localhost/repo/?mod=view-license&upload=235&item=494099
```

**Comment by GMishx** [comment]
Oops, looks like max-width and height needs to be removed from `.modal` class
snap

File: src/composer.json [hunk 1]
```diff
@@ -28,7 +28,8 @@
         "twig/extensions" : "v1.5.4",
         "guzzlehttp/guzzle": "v6.5.5",
         "container-interop/container-interop": "v1.2.0",
-        "phpoffice/phpspreadsheet": "^1"
+        "phpoffice/phpspreadsheet": "^1",
+        "twbs/bootstrap": "4.0.0"
```
Comment (GMishx) [inline src/composer.json:32]: Possible to use latest version `v4.6.0`?

**Comment by shaheemazmalmmd** [comment]
> Oops, looks like max-width and height needs to be removed from `.modal` class
> snap

Im not able to reproduce this issue, tried it in Mozilla and Chrome everything looks fine. can you please retest.

File: src/composer.json (hunk 1 above)
Comment (shaheemazmalmmd) [reply src/composer.json:32]: Update requires many changes. i think lets merge this as initial commit and we can update it in future if required.

**Comment by GMishx** [comment]
> Im not able to reproduce this issue, tried it in Mozilla and Chrome everything looks fine. can you please retest.

I uninstalled fossology and installed the branch again and the issue was gone :man_shrugging: .

However, there is a new issue with Notice modal which I believe can be fixed in a separate PR before tagging next release

image


Also, one general feedback. Will it be possible to make the bulk modal not block the background interactions so user can easily copy texts from file for pasting in ref text and license/acknowledgement text?

File: src/composer.json (hunk 1 above)
Comment (GMishx) [reply src/composer.json:32]: Sure.

**Review by GMishx — APPROVED** [review]
Changes good.

Tested, working fine.

**Pull request merged**